### PR TITLE
resample audio to 44.1 khz

### PR DESCRIPTION
**Pull request by aniketmaurya**
We can resample to the required sample rate rather than raising an error.

### CODE REVIEW

**Comment by vb000** [comment]
Hi @aniketmaurya,

Thanks for the very useful PR! I see a small issue with the current version: the output is saved at `fs` sampling rate, but the true sampling rate of the output audio would be 44.1 khz. Following patch seems to work:

```diff
diff --git a/Waveformer.py b/Waveformer.py
index 72e5b84..bc198fb 100644
--- a/Waveformer.py
+++ b/Waveformer.py
@@ -47,13 +47,14 @@ if __name__ == '__main__':
     # Instantiate model
     params = utils.Params('default_config.json')
     model = Waveformer(**params.model_params)
-    utils.load_checkpoint('default_ckpt.pt', model)
+    model.load_state_dict(torch.load(
+        'default_ckpt.pt', map_location=torch.device('cpu'))['model_state_dict'])
     model.eval()
 
     # Read input audio
     mixture, fs = torchaudio.load(args.input)
-    # assert fs == 44100, "Input sampling rate must be 44.1 khz."
-    mixture = torchaudio.functional.resample(mixture, orig_freq=fs, new_freq=44100)
+    if fs != 44100:
+        mixture = torchaudio.functional.resample(mixture, orig_freq=fs, new_freq=44100)
     mixture = mixture.unsqueeze(0)
     print("Loaded input audio from %s" % args.input)
 
@@ -66,8 +67,10 @@ if __name__ == '__main__':
             query[0, TARGETS.index(t)] = 1.
 
     with torch.no_grad():
-        output = model(mixture, query)
+        output = model(mixture, query).squeeze(0)
+    if fs != 44100:
+        output = torchaudio.functional.resample(output, orig_freq=44100, new_freq=fs)
     print("Inference done. Saving output audio to %s" % args.output)
 
     assert not os.path.exists(args.output), "Output file already exists."
-    torchaudio.save(args.output, output.squeeze(0), fs)
+    torchaudio.save(args.output, output, fs)
```

I included an unrelated fix here (moving weights to the CPU) for running the model on CPU only machines. You can either include it or discard it in this PR.

File: Waveformer.py [hunk 1]
```diff
@@ -1,58 +1,103 @@
...
+        ckpt_url = "https://targetsound.cs.washington.edu/files/default_ckpt.pt"
         print("\nDownloading the checkpoint from %s:" % ckpt_url)
         wget.download(ckpt_url)
 
     # Instantiate model
-    params = utils.Params('default_config.json')
+    params = utils.Params("default_config.json")
     model = Waveformer(**params.model_params)
-    utils.load_checkpoint('default_ckpt.pt', model)
+    if torch.cuda.is_available():
```
Comment (aniketmaurya) [inline Waveformer.py:88]: this will help in running the model on CUDA whenever available otherwise on CPU

File: Waveformer.py [hunk 2]
```diff
@@ -62,11 +107,13 @@
     else:
         query = torch.zeros(1, len(TARGETS))
         for t in args.targets:
-            query[0, TARGETS.index(t)] = 1.
+            query[0, TARGETS.index(t)] = 1.0
 
-    with torch.no_grad():
-        output = model(mixture, query)
+    with torch.inference_mode():
```
Comment (aniketmaurya) [inline Waveformer.py:112]: inference mode is recommended https://discuss.pytorch.org/t/pytorch-torch-no-grad-vs-torch-inference-mode/134099/2?u=timgianitsos

**Comment by aniketmaurya** [comment]
Thanks a lot @vb000 for taking your time to review this PR 😄 

I have applied your suggestions, along with some code formatting using [black](https://github.com/psf/black).

**Pull request merged**